libs/remote_storage/src/s3_bucket.rs

```diff
@@ -158,12 +154,24 @@ impl S3Bucket {
             }
             prefix
         });
+
+        let rps = aws_config.concurrency_limit.get();
+        let concurrency_limiter = leaky_bucket::RateLimiter::builder()
+            .max(rps)
+            .initial(0)
```

Comment:
Found in testing; zero here means that first tenant loads happen at around 1s - 20ms. Unsure what my initial idea was, but this needs to be fixed.